src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75d;
+    private static final int GROW_NUMBER = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            Node<K,V> curNode = table[index];
+            while (curNode != null) {
+                if (Objects.equals(curNode.key, key)) {
+                    curNode.value = value;
+                    return;
+                }
+                if (curNode.next == null) {
+                    curNode.next = new Node<>(key, value, null);
+                    break;
+                }
+                curNode = curNode.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> curNode = table[getIndex(key)];
+        while (curNode != null) {
+            if (Objects.equals(curNode.key, key)) {
+                return curNode.value;
+            }
+            curNode = curNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHashCode(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode());
+    }
+
+    private int getIndex(K key) {
+        return getHashCode(key) % table.length;
+    }
```

Comment:
you can merge this two methods, because `getHashCode()` quite small and called only one time in the code.